### ISSUE TITLE
LoadError: UndefVarError: Serial not defined

### ISSUE THREAD

**Issue by BoGGoG** [lucass-carneiro/QuasinormalModes.jl#2]
Hi,
thanks for the cool package.

I get the error
```julia
ERROR: LoadError: UndefVarError: Serial not defined
Stacktrace:
 [1] top-level scope
   @ ~/Documents/tmp/QuasinormalModes.jl/examples/harmonic_oscilator.jl:68
 [2] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [3] top-level scope
   @ REPL[2]:1
in expression starting at /Users/myuser/Documents/tmp/QuasinormalModes.jl/examples/harmonic_oscilator.jl:68
```

I didn't do much, I installed the package, cloned the repo and ran the file in Julia.

I'm also pretty new to Julia, so if I'm just too dump then I'm sorry.

**Comment by BoGGoG** [comment]
Ok, going through the commits I saw that `Serial` and `Threaded` have been added recently. I think the package loaded from `pkg> add QuasinormalModes` might still be the one without those features and thus lines like
```julia
m_ana = computeEigenvalues(Serial(), p_ana, c_ana)
```
cannot work. Just removing `Serial()` in the line above helped.

Checking the versions also confirms this:
```julia
(@v1.6) pkg> status QuasinormalModes
      Status `~/.julia/environments/v1.6/Project.toml`
  [7b4b3212] QuasinormalModes v0.2.1
```

whereas the GitHub version is v0.3.0.

To get the new version I tried: `pkg> add https://github.com/lucass-carneiro/QuasinormalModes.jl`
which gave the error
```julia
(@v1.6) pkg> add https://github.com/lucass-carneiro/QuasinormalModes.jl
     Cloning git-repo `https://github.com/lucass-carneiro/QuasinormalModes.jl`
    Updating git-repo `https://github.com/lucass-carneiro/QuasinormalModes.jl`
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package FastTransforms [057dd010]:
 FastTransforms [057dd010] log:
 ├─possible versions are: 0.5.0-0.12.4 or uninstalled
 ├─restricted by compatibility requirements with ApproxFun [28f2ccd6] to versions: 0.5.0-0.12.4
 │ └─ApproxFun [28f2ccd6] log:
 │   ├─possible versions are: 0.9.0-0.12.5 or uninstalled
 │   └─restricted to versions * by an explicit requirement, leaving only versions 0.9.0-0.12.5
 └─restricted by compatibility requirements with DSP [717857b8] to versions: uninstalled — no versions left
   └─DSP [717857b8] log:
     ├─possible versions are: 0.5.1-0.7.1 or uninstalled
     └─restricted by compatibility requirements with Polynomials [f27b6e38] to versions: 0.7.0-0.7.1 or uninstalled
       └─Polynomials [f27b6e38] log:
         ├─possible versions are: 0.5.0-2.0.12 or uninstalled
         └─restricted to versions 2 by QuasinormalModes [7b4b3212], leaving only versions 2.0.0-2.0.12
           └─QuasinormalModes [7b4b3212] log:
             ├─possible versions are: 0.3.0 or uninstalled
             └─QuasinormalModes [7b4b3212] is fixed to version 0.3.0
```

Not sure if this should be a separate issue.

**Comment by lucass-carneiro** [comment]
Hi there! I'm sorry for the inconvenience, and hope that you find the package useful in the future.

Indeed, the `Serial()` and `Threaded()` options are available from version `0.3.0` onward, but this is the current version available in the Julia package registry, so you should have been able to automatically install it directly with `PKG` instead of manually adding the Git repository.

From the error message, it looks like my package dependency on certain versions of `Polynomials` cascaded into an unsatisfied dependency of the package `FastTransforms`, which I don't even use directly. Here's what I would do:

1. Remove the currently installed or partially installed versions of `QuasinormalModes`. You can do that by entering the `PKG` REPL and typing `remove QuasinormalModes`.
2. Remove the build and source files. This step is just to make sure that we completely removed the package. In your home directory there should be a `.julia` folder. What you want to do is delete the folder `QuasinormalModes` from `~/.julia/packages` and  `~/.julia/compiled/` if they exist.
3. In the `PKG` REPL, issue the `update` command. This should update all your currently added packages to their newest versions. I'm guessing your issue might have occurred because of older packages, but that's mainly a guess.
4. After updating Julia may issue a warning saying it could not precompile all packages and a restart is needed. In that case, exit the Julia REPL and load it again. Enter the `PKG` REPL and issue `precompile`. The packages that couldn't be precompiled before should now compile.
5. After everything is up to date and precompiled you can issue `add QuasinormalModes` in the `PKG` REPL again and hopefully succeed in installing the package. I personally like to quit Julia  after step 4 and load it again before issuing `add`.

Whenever I want to add a new package in Julia, I do the following:

1. Enter the `PKG` REPL, and issue `update`.
2. After a successful update, I quit Julia and start it again. Only then I issue the `add` command.
3. If the precompilation restart warning is issued, I restart Julia and issue `precompile` in the `PKG` REPL. I restart julia after this and only then I issue `add`.

These steps might not be required, but I found that this works consistently in all my Julia installations. I have currently 2 Julia versions, `1.6.0` and `1.6.1`. Unfortunately I was unable to reproduce your problem in neither of them. In both versions, I first removed the package, updated everything and added it again using the steps I described above. My version `1.6.0` was quite out of date so the manual `precompile` was required in this version. Both installations went of without problems and checking the installed version with `status QuasinormalModes` in the `PKG` REPL confirmed that version `0.3.0` was installed automatically, as it should be.

If you continue to encounter problems please fell free to contact me and I shall do my best to try and help you out. It may be very well possible that there is a problem with my package itself and how I specify the versions for the dependencies.

**Comment by BoGGoG** [comment]
Thank you so much for the answer!
